x-pack/apm-server/sampling/config.go

```diff
@@ -153,90 +155,115 @@ type PolicyCriteria struct {
 	TraceName string
 }
 
+var (
+	ErrBeatIDMissing                = errors.New("BeatID unspecified")
+	ErrBashProcessorMissing         = errors.New("BatchProcessor unspecified")
+	ErrFlushIntervalInvalid         = errors.New("FlushInterval unspecified or negative")
+	ErrMaxDynamicServicesInvalid    = errors.New("MaxDynamicServices unspecified or negative")
+	ErrPoliciesMissing              = errors.New("Policies unspecified")
+	ErrDefaultPolicyMissing         = errors.New("Policies does not contain a default (empty criteria) policy")
+	ErrIngestRateDecayFactorInvalid = errors.New("IngestRateDecayFactor unspecified or out of range (0,1]")
+	ErrCompressionLevelInvalid      = errors.New("CompressionLevel out of range [-1,9]")
+	ErrElasticsearchMissing         = errors.New("Elasticsearch unspecified")
+	ErrDBMissing                    = errors.New("DB unspecified")
+	ErrStorageDirMissing            = errors.New("StorageDir unspecified")
+	ErrStorageGCIntervalInvalid     = errors.New("StorageGCInterval unspecified or negative")
+	ErrTTLInvalid                   = errors.New("TTL unspecified or negative")
+	ErrSampleRateInvalid            = errors.New("SampleRate unspecified or out of range [0,1]")
```

Comment:
I'm not too keen on exporting these errors. I don't think there's any good reason for a client of this package to check the specific error value.